### PR TITLE
reduce re-render on userDashboard

### PR DESCRIPTION
**Pull request by Sbonelo01**
Related issues: [RENDER tidy]

- [x] My code follows the style guidelines of this project
- [x] I have merged the develop branch into my branch and fixed any merge conflicts
- [x] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by TheMaja — APPROVED** [review]

**Review by Vuyisanani — APPROVED** [review]

**Review by sheenarbw — CHANGES_REQUESTED** [review]
When in doubt: don't guess.

Here is the correct mapStateToProps:

```

const mapStateToProps = (state) => {
  return {
    users: state.apiEntities.users,
    userDetailedStats: state.apiEntities.userDetailedStats,
    userBurndownStats: state.apiEntities.burndownSnapshots,
    authUser: state.App.authUser
  };
};
```

Dear reviewers: please make sure that the code you are reviewing actually does what it is meant to do. It's very important.  This PR doesn't address the issue it's meant to address at all.

**Review by KhalidUmuzi — CHANGES_REQUESTED** [review]
Please make the requested changes as Sheena mentioned. This is still not correct if i look at the whole file.

lines 48-56 are still the same at the moment

**Review by KhalidUmuzi — CHANGES_REQUESTED** [review]
Please fix Merge conflicts

File: frontend/src/components/pages/UserDashboard/index.jsx [hunk 1]
```diff
@@ -36,14 +48,15 @@ function DashboardUnconnected({
...
 }
 
 const mapStateToProps = (state) => {
   return {
-    users: state.apiEntities.users || {},
-    userDetailedStats: state.apiEntities.userDetailedStats || {},
+    users: state.apiEntities.users,
+    userDetailedStats: state.apiEntities.userDetailedStats,
+    userBurndownStats: state.apiEntities.burndownSnapshots,
     // authedUserId: state.App.authUser.userId,
```
Comment (KhalidUmuzi) [inline frontend/src/components/pages/UserDashboard/index.jsx]: Something wierd is going on here. I don't think the burndown stats is on the dashboard any longer. Its been moved to UserActions That change has been merged into develop already.

We don't want to add it back. I think this file might have been changed when you were looking at something

**Review by KhalidUmuzi — CHANGES_REQUESTED** [review]

**Review by KhalidUmuzi — APPROVED** [review]

File: frontend/src/components/pages/UserDashboard/index.jsx [hunk 2]
```diff
@@ -42,10 +48,11 @@ function DashboardUnconnected({
 
 const mapStateToProps = (state) => {
   return {
-    users: state.apiEntities.users || {},
-    userDetailedStats: state.apiEntities.userDetailedStats || {},
+    users: state.apiEntities.users,
+    userDetailedStats: state.apiEntities.userDetailedStats,
+    userBurndownStats: state.apiEntities.burndownSnapshots,
```
Comment (sheenarbw) [inline frontend/src/components/pages/UserDashboard/index.jsx]: Why is userBurnDownStats here?  Wasn't it removed on purpose?

If you search for userBurnDownStats in this file then you will see that it is not being used.

File: frontend/src/components/pages/UserDashboard/index.jsx [hunk 3]
```diff
@@ -66,6 +73,18 @@ const mapDispatchToProps = (dispatch) => {
         })
       );
     },
+
+    //TODO Implement Page check
+    fetchUserBurndownStats: ({ userId }) => {
```
Comment (sheenarbw) [inline frontend/src/components/pages/UserDashboard/index.jsx]: Please also remove this function. It looks like it isn't being used either.

**Review by sheenarbw — CHANGES_REQUESTED** [review]
I'm not sure why Khalid's comment wasn't addressed.

**Review by KhalidUmuzi — APPROVED** [review]

**Review by sheenarbw — APPROVED** [review]

**Pull request merged**